dandi/bids_validator_xs.py

```diff
@@ -453,7 +467,7 @@ def validate_all(
 
 def write_report(
     validation_result,
-    report_path="{logdir}/bids-validator-report_{datetime}-{pid}.log",
+    report_path="/var/tmp/bids-validator/report_{datetime}-{pid}.log",
```

Comment:
looks wrong.
So this is the code which is borrowed from 3rd party (upstream `bids-specification`)?   could such change be introduced "upstream"?

A possible workflow for carrying 3rd party components to make it clear(er) on what changes we introduce here and carry them through the upgrades of the thirdparty code (we used it in PyMVPA)

- keep all thirtd-party code in a consistent descriptive location, e.g. `dandi.support.third_party` (IIRC submodule cannot start with `3` so we will use `third`)
- establish "third-party" branch at some point where pristine versions (no modifications) of thirdparty code are introduced. (that branch could even have no common base with main tree, just carry that `dandi/support/third_party` folder)
- introduce new versions of third-party **only** in that branch. merge that branch into `master` when that is done. Then any changes to upstream code would merge with our code

as we hope that all of this would be a legit pypi package soon, may be "too late" for us here, but please first inspect `git log` on 3rd party content to see if any other change was "dropped" (has some commit what was not migrated upstream)

Reply:
set it up in a separate submodule for now. A couple of changes are still pending upstream in bids-specification.

Reply:
> set it up in a separate submodule for now

git-rotten mind of me immediately thought about `git submodules`. I guess we could have used that one as well (instead of branching mechanism I described above), but I guess could be avoided.



> A couple of changes are still pending upstream in bids-specification.

I guess 
- https://github.com/bids-standard/bids-specification/pull/1120
- https://github.com/bids-standard/bids-specification/pull/1122
- https://github.com/bids-standard/bids-specification/pull/1138
?  seems all could be "adopted" merged in here later , we should not wait for them needing to amass even more changes, let's finish/merge this PR to the current state of things in BIDS validation land.